repos/system_upgrade/common/actors/checkleftoverpackages/libraries/checkleftoverpackages.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from leapp.libraries.common.config.version import get_source_major_version
+from leapp.libraries.common.rpms import get_installed_rpms, get_leapp_dep_packages, get_leapp_packages
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledUnsignedRPM, LeftoverPackages, RPM
+
+
+def process():
+    LEAPP_PACKAGES = get_leapp_packages(major_version=[get_source_major_version()])
+    LEAPP_DEP_PACKAGES = get_leapp_dep_packages(major_version=[get_source_major_version()])
+    installed_rpms = get_installed_rpms()
+
+    if not installed_rpms:
+        return
+
+    leftover_pkgs_to_remove = LeftoverPackages()
+    unsigned = [pkg.name for pkg in next(api.consume(InstalledUnsignedRPM), InstalledUnsignedRPM()).items]
+
+    for rpm in installed_rpms:
+        rpm = rpm.strip()
+        if not rpm:
+            continue
+        name, version, release, epoch, packager, arch, pgpsig = rpm.split('|')
```

Comment:
+1 was thinking about the same thing, however, this is just a copied file from our code and the PR is about unit tests, so I don't know if it should be fixed here. If we did not hit it until now then it's probably safe. Although, I would still fix it here, while we still have it on the radar.

Reply:
btw, we have about that already bug :) as one company put the `|` character into the packager section :) so we will be changing it to something else unfortunately. even when this can be considered as super corner-case

Reply:
it can be still used, but definitely catch the exception and log the problematic line. actors after the rpm upgrade transaction must not crash

Reply:
Implemented catching error and priting warnmsg, with printing the problematic rpm.